js/views/userPage/TypeFilter.js

```diff
@@ -0,0 +1,91 @@
+/* categoryFilter.js and typeFilter.js share quite a bit. Ensure that when one is updated,
+the other is also maintained.
+*/
+
+import $ from 'jquery';
+import _ from 'underscore';
+import loadTemplate from '../../utils/loadTemplate';
+import BaseVw from '../baseVw';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    const opts = {
+      ...options,
+    };
+
+    opts.initialState = {
+      types: [],
+      selected: 'all',
+      expanded: false,
+      maxInitiallyVisibleTypes: 6,
+      ...(options.initialState || {}),
+    };
+
+    super(opts);
+    this.options = opts;
+    this._state = opts.initialState;
```

Comment:
This is not needed. Since the initial state is being passed into `super`, the `baseVw` constructor will take care of this.

Reply:
ah @jashot7, I see what you did. You were just copying the CategoryFilter which was made before all this stuff was moved into the baseVw. If you have some time, I'd just delete the overridden methods (and replaceState) and see if it works with the baseVw. It probably will and would make the code-base cleaner.